src/main/java/gov/nasa/cumulus/CNMResponse.java

```diff
@@ -105,8 +105,6 @@ public static String generateOutput(String inputCnm, String exception, JsonObjec
 
 		JsonElement sizeElement = inputKey.get("product").getAsJsonObject().get("files").getAsJsonArray().get(0).getAsJsonObject().get("size");
 
-		inputKey.remove("product");
```

Comment:
We need to modify uri in each files object because we want uri in the CNM response to point to the direct download link of the file in the archive. For example, https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MODIS_A-JPL-L2P-v2019.0/20020704000015-JPL-L2P_GHRSST-SSTskin-MODIS_A-N-v02.0-fv01.0.nc

However, modifying the uri is still not enough because for missions such as SWOT the SDS points us to a single tar file in the original CNM and we uncompress and archive the individual files. In the CNM-R we want the files list to contain all the individual files. It may make more sense to build product.files from the Cumulus granule object.

Also, product.name should not be the product.name from the original CNM. It should point to the granuleId because we determine the granule ID during ingest which isn't always the same as the provider product.name.

Reply:
Have replaced all files objects with distribution_endpoint leading html (which also requires a terraform change in cumulus-deploy-tf project - will follow) 
product.name is replaced by granuleId

unit tests all passed.  Sanity unit test case removed because it is well covered by the success and failure cases.